### PR TITLE
feat: create endpoint to check if user has already opened position 

### LINKED ISSUE #210: [Backend] Create endpoint to check if user has already opened position
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Add endpoint `has_user_opened_position` GET in this [file](https://github.com/djeck1432/spotnet/blob/main/web_app/api/user.py)
3. This method should have as input `wallet_id` and check in db if user has opened position or not.
4. Create crud method to get `boolean` value if user has opened position in [PositionDBConnector](https://github.com/djeck1432/spotnet/blob/main/web_app/db/crud.py#L209)
5. Return value `true` or `false` in response of endpoint.

### PR DESCRIPTION
**Pull request by dahtee**
### Changes Made
- Added new GET endpoint `/api/has-user-opened-position` to check user's position status
- Implemented CRUD method `has_opened_position` in PositionDBConnector
- Added boolean response for position status check

### Technical Details
- New endpoint accepts `wallet_id` as input parameter
- Returns simple boolean response: `true` if user has opened position, `false` otherwise
- Implements proper error handling for database queries

### Related Issue
Closes #210

### CODE REVIEW

**Comment by dahtee** [comment]
Hello @djeck1432 kindly review.

File: web_app/api/user.py [hunk 1]
```diff
@@ -23,6 +23,28 @@
...
+    :param wallet_id: wallet id
+    :return: Dict containing boolean result
+    :raises: HTTPException
+    """
+    try:
+        has_position = position_db.has_opened_position(wallet_id)
+        return {"has_opened_position": has_position}
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
```
Comment (djeck1432) [inline web_app/api/user.py:44]: 500 error related to server error, 
it should be 4** status response

File: web_app/api/user.py [hunk 2]
```diff
@@ -23,6 +23,28 @@
...
+    """
+    Check if a user has any opened positions.
+    :param wallet_id: wallet id
+    :return: Dict containing boolean result
+    :raises: HTTPException
+    """
+    try:
+        has_position = position_db.has_opened_position(wallet_id)
+        return {"has_opened_position": has_position}
+    except Exception as e:
```
Comment (djeck1432) [inline web_app/api/user.py:42]: Please, be more specific with exception, try to avoid using generic one

File: web_app/db/crud.py [hunk 3]
```diff
@@ -241,6 +241,32 @@ def _get_user_by_wallet_id(self, wallet_id: str) -> User | None:
...
+        :param wallet_id: str
+        :return: bool
+        """
+        with self.Session() as db:
+            user = self._get_user_by_wallet_id(wallet_id)
+            if not user:
+                return False
+
+            try:
+                position_exists = (
```
Comment (djeck1432) [inline web_app/db/crud.py:256]: Please, use already exisiting method `exists`

```
position_exists = db.query(
    db.query(Position)
    .filter(
        Position.user_id == user.id,
        Position.status == Status.OPENED.value,
    )
    .exists()
).scalar()
```

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by dahtee** [comment]
On it now Sir.

**Comment by djeck1432** [comment]
@dahtee Please, fix pylint issue

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**